invenio_access/permissions.py

```diff
@@ -76,13 +90,20 @@ def _load_permissions(self):
                 ).all()
 
                 for actionrole in actionsroles:
-                    result[actionrole.exclude].add(
+                    action[actionrole.exclude].add(
                         actionrole.need
                     )
+
+                result[True] |= action[True]
+                result[False] |= action[False]
+
+                if cache:
+                    cache.set(cache_key, action)
+
             else:
                 result[self.NEEDS].add(explicit_need)
 
-        self._permissions = result
```

Comment:
It depends how long you keep the instance of `DynamicPersmission`. Hence we should document that it's evaluated when you instantiate it.